Rename script + add support for Google Compute Engine

jkaplowitz opened the pull request:
_(The Debian wiki's existing documentation will probably need to be updated together with this merge, to replace former references to the script name `ec2debian-build-ami` with `build-debian-cloud` or `build-debian-cloud ec2` as appropriate.)_

This commit is the result of collaboration among David McWherter, Anders
Ingemann, and me. The rename to build-debian-cloud is at Anders's
initiative.

David's and my work on this is Copyright 2013 Google, Inc. and is
released under the Apache License 2.0.

jkaplowitz committed:
Add support for Google Compute Engine

This commit is the result of collaboration among David McWherter, Anders
Ingemann, and me. The rename to build-debian-cloud is at Anders's
initiative.

David's and my work on this is Copyright 2013 Google, Inc. and is
released under the Apache License 2.0.

tmatilai commented on plugins/README.md:
```diff
@@ -28,8 +28,5 @@ The following is a list of external plugins you can use with ec2debian-build-ami
 
-* [ec2debian-chef](https://github.com/tmatilai/ec2debian-chef)  
-  Installs [Opscode Chef](http://www.opscode.com/chef/) client using the Omnibus installer.
-
```
Hey, please do not remove the reference to my mighty plugin! =)
Actually it should be generic enough to work also with gce. So it could be renamed, too.

jkaplowitz replied:
Hi - I didn't mean to remove that. It was done accidentally when rebasing against Anders' tree (most of our internal work was before he merged your PR). I'll push a commit shortly to fix that, and maybe another small thing or two.

tmatilai replied:
Yeah no prob. I guessed that. :)

jkaplowitz replied:
Revised the PR with this fix and with a different default volume size for Google Compute Engine (in line with other Google-provided images).

jkaplowitz committed:
Re-add accidentally removed plugins/README.md text

jkaplowitz committed:
Different volume_size defaults for EC2 and Google

In keeping with Google Compute Engine's other images, default to 10 GB
for our Debian images. Image size is handled differently for Amazon EC2,
so leave the pre-existing 1 GB default alone there.

Also update the comment in the affected build-debian-cloud section to
match reality (the --filesystem option already exists).

jkaplowitz committed:
Remove /etc/resolv.conf during image creation

The build system's DNS info should not pollute the image. DHCP will
recreate this for the VM environment.

Also renumber tasks/gce/52-* to tasks/gce/25-* so that those steps occur
when DNS is still functional inside the image.

jkaplowitz committed:
Clean up loopback devices and mounts on all exits

Previously, depending on where a Google Compute Engine image build
failed, the loopback mount could linger, which is cumbersome. This
cleans that up, though it leaves the workspace around during unclean
exits to facilitate debugging.

jkaplowitz committed:
Adjust two traps to co-exist

Turns out a pre-existing use of 'trap' in Anders' code stepped on the
trap I added. This coordinates, in a way that shouldn't break non-Google
builds.
